executor.go

```diff
@@ -397,7 +397,10 @@ func doesFragmentConditionMatch(eCtx *ExecutionContext, fragment ast.Node, ttype
 		if conditionalType == ttype {
 			return true
 		}
-
+                if conditionalType.Name() == ttype.Name() {
```

Comment:
@pyros2097 - I might be missing something, do you really think it should be rely only on `.Name` to say are the same types?